packages/flutter/lib/src/foundation/collections.dart

```diff
@@ -4,16 +4,16 @@
 
 // TODO(ianh): These should be on the Set and List classes themselves.
 
-/// Compares two sets for deep equality.
+/// Compares two sets for shallow equality.
 ///
 /// Returns true if the sets are both null, or if they are both non-null, have
 /// the same length, and contain the same members. Returns false otherwise.
 /// Order is not compared.
 ///
-/// The term "deep" above refers to the first level of equality: if the elements
-/// are maps, lists, sets, or other collections/composite objects, then the
-/// values of those elements are not compared element by element unless their
-/// equality operators ([Object.==]) do so.
+/// If the elements in the sets are collections (e.g., maps, lists, sets),
+/// then the result may not be as expected since the function uses the
```

Comment:
I am not sure this adds clarity to this. What might be expected? 
Also, does the see also for lists and maps not provide additional guidance?

Reply:
Hi @Piinks, Thank you for your review. The root of my problem was the "deep" word in the first line of the documentation. When I first used this function, I only read the first line and expected a recursive check for equality.

You are right, if the first line is changed to shallow equality, then the expected behavior is different, and so these lines can be confusing.

See also refers to the "twin" functions for lists and maps where the problem is the same.

@gspencergoog also reviewed the changes, so I will make a new commit soon based on your comments.